File: examples/send-message.py
#!/usr/bin/env python3
"""
Example: Send a text message via Omdaa API (Python)

Run from repo root after install:
  pip install -e packages/omdaa-python
  export OMDAA_API_KEY=your-api-key
  export OMDAA_TO=9665XXXXXXXX   # optional
  python3 examples/send-message.py

Or from your project after: pip install omdaa-api-client
"""

import os
import sys

# From repo: make package available from packages/omdaa-python
_repo_root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
_src = os.path.join(_repo_root, "packages", "omdaa-python", "src")
if os.path.isdir(_src) and _src not in sys.path:
    sys.path.insert(0, _src)

from omdaa import OmdaaClient, OmdaaError

def main():
    api_key = os.environ.get("OMDAA_API_KEY", "")
    to = os.environ.get("OMDAA_TO", "966500000000")

    if not api_key:
        print("Set OMDAA_API_KEY then run the script again.", file=sys.stderr)
        sys.exit(1)

    client = OmdaaClient(api_key)

    try:
        result = client.messages.send_text({
            "sessionId": "default",
            "to": to,
            "message": "Hello from Omdaa SDK (Python)",
        })
        print("Sent successfully:", result)
    except OmdaaError as e:
        print("API error:", e.status_code, str(e), file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()

File: examples/send-message.js
#!/usr/bin/env node
/**
 * مثال جاهز: إرسال رسالة نصية عبر Omdaa API (Node.js)
 *
 * التشغيل من جذر المستودع بعد التثبيت والبناء:
 *   make install   (أو: cd packages/omdaa-js && npm install && npm run build)
 *   export OMDAA_API_KEY=your-api-key
 *   export OMDAA_TO=9665XXXXXXXX   # اختياري
 *   node examples/send-message.js
 *
 * أو من مشروعك بعد: npm install omdaa-api-client
 */

const path = require('path');
let OmdaaClient, OmdaaError;
try {
  ({ OmdaaClient, OmdaaError } = require('omdaa-api-client'));
} catch {
  const localPath = path.resolve(__dirname, '../packages/omdaa-js/dist/index.js');
  ({ OmdaaClient, OmdaaError } = require(localPath));
}

const apiKey = process.env.OMDAA_API_KEY;
const to = process.env.OMDAA_TO || '966500000000';

if (!apiKey) {
  console.error('عيّن OMDAA_API_KEY ثم شغّل السكربت مرة أخرى.');
  process.exit(1);
}

const client = new OmdaaClient({ apiKey });

async function main() {
  try {
    const res = await client.messages.sendText({
      sessionId: 'default',
      to,
      message: 'مرحباً من Omdaa SDK (Node.js)',
    });
    console.log('تم الإرسال بنجاح:', res.data?.messageId || res);
  } catch (err) {
    if (err instanceof OmdaaError) {
      console.error('خطأ من API:', err.status, err.message);
    } else {
      console.error(err);
    }
    process.exit(1);
  }
}

main();
